Mathlib/Algebra/Category/ModuleCat/OverFilteredColimits.lean

```diff
@@ -0,0 +1,351 @@
+  have eq₃ : m₁ + m₂ = colimit.ι ℳ j
+      (ℳ.map (IsFiltered.toSup O H $ by simp [O]) (Concrete.repColimit ℳ m₁) +
+       ℳ.map (IsFiltered.toSup O H $ by simp [O]) (Concrete.repColimit ℳ m₂)) := by
+    rw [map_add]
+    erw [colimit.w_apply, colimit.w_apply, Concrete.ι_repColimit_eq, Concrete.ι_repColimit_eq]
+
+  rw [eq₃]
+  conv_rhs => rw [eq₁]; rhs; rw [eq₂]
+  rw [smul_rep, smul_rep, smul_rep, hsmul.smul_add, map_add]
+
+lemma add_smul (c : J) (r₁ r₂ : ℛ.obj c) (m : colimit (C := 𝔄𝔟) ℳ) :
+    smulColimit (r₁ + r₂) m = smulColimit r₁ m + smulColimit r₂ m := by
+  rw [show m = colimit.ι ℳ (Concrete.indexRepColimit ℳ m) _ by
+    erw [Concrete.ι_repColimit_eq], smul_rep, hsmul.add_smul, smul_rep, smul_rep, map_add]
+
+lemma zero_smul (c : J) (m : colimit (C := 𝔄𝔟) ℳ) :
+    smulColimit (ℳ := ℳ) (0 : ℛ.obj c) m = 0 := by
+  rw [show m = colimit.ι ℳ (Concrete.indexRepColimit ℳ m) _ by
+    erw [Concrete.ι_repColimit_eq], smul_rep, hsmul.zero_smul, map_zero]
+
+end smulColimit
+
+noncomputable instance moduleObjColimit (j : J) :
+    Module (ℛ.obj j) (colimit (C := 𝔄𝔟) ℳ) where
+  smul := smulColimit
+  one_smul := smulColimit.one_smul _ _ _
+  mul_smul := smulColimit.mul_smul _ _ _
+  smul_zero := smulColimit.smul_zero _ _ _
+  smul_add := smulColimit.smul_add _ _ _
+  add_smul := smulColimit.add_smul _ _ _
+  zero_smul := smulColimit.zero_smul _ _ _
+
+variable {ℛ ℳ} in
+/--
+Let `R` be the filtered colimit of rings `{Rⱼ}` and `M` be the filtered colimit of
+abelian groups `{Mⱼ}`  with the same indexing set `j ∈ J`, if for each `j ∈ J`, `Mⱼ` is an `Rⱼ` such
+that the `Rⱼ`-action is compatible, then there is a scalar multiplication
+`R → M → M`.
+-/
+noncomputable def colimitSMulColimit (r : colimit (C := ℜ𝔦𝔫𝔤) ℛ) (m : colimit (C := 𝔄𝔟) ℳ) :
+    colimit (C := 𝔄𝔟) ℳ :=
+  (smulColimit (Concrete.repColimit ℛ r) m)
+
+namespace colimitSMulColimit
+
+lemma rep_smul {c : J} (r : ℛ.obj c) (m : colimit (C := 𝔄𝔟) ℳ) :
+    colimitSMulColimit (colimit.ι ℛ c r) m = smulColimit r m := by
+  delta colimitSMulColimit
+  rw [show m = colimit.ι ℳ (Concrete.indexRepColimit ℳ m) _ by
+    erw [Concrete.ι_repColimit_eq], smulColimit.smul_rep]
+  apply hsmul.respect_ι
+  · erw [Concrete.ι_repColimit_eq]
+  · erw [Concrete.ι_repColimit_eq, Concrete.ι_repColimit_eq]
+
+protected lemma one_smul (m : colimit (C := 𝔄𝔟) ℳ) :
+    colimitSMulColimit (1 : colimit (C := ℜ𝔦𝔫𝔤) ℛ) m = m := by
+  let c : J := (inferInstance : IsFiltered J).2.some
+  rw [show (1 : colimit (C := ℜ𝔦𝔫𝔤) ℛ) = colimit.ι ℛ c 1 by
+    rw [map_one], rep_smul, smulColimit.one_smul]
+
+protected lemma mul_smul
+    (r₁ r₂ : colimit (C := ℜ𝔦𝔫𝔤) ℛ) (m : colimit (C := 𝔄𝔟) ℳ) :
+    colimitSMulColimit (r₁ * r₂) m = colimitSMulColimit r₁ (colimitSMulColimit r₂ m) := by
+  classical
+  let O : Finset J :=
+    {  Concrete.indexRepColimit ℛ r₁, Concrete.indexRepColimit ℛ r₂ }
+  let H : Finset ((X : J) ×' (Y : J) ×' (_ : X ∈ O) ×' (_ : Y ∈ O) ×' (X ⟶ Y)) := {}
```

Comment:
You can remove H and replace all the occurrences with `{}` if you like.

Reply:
done